packages/fxa-auth-server/lib/routes/account.ts

```diff
@@ -643,6 +645,73 @@ export class AccountHandler {
     }
   }
 
+  async setPassword(request: AuthRequest) {
+    this.log.begin('Account.set_password', request);
+
+    const form = request.payload as any;
+    const { authPW, metricsContext } = form;
+    const { query } = request;
+    const auth = request.auth;
+    const { user: uid } = auth.credentials;
+
+    const account = await this.db.account(uid);
+    const email = account.primaryEmail.email;
+
+    await this.customs.check(request, email, 'setPassword');
+
+    const response: Record<string, any> = {};
+    response.uid = uid;
+
+    try {
+      await this.setPasswordOnStubAccount(account, authPW);
+
+      const { emailCode: tokenVerificationId } = account;
+      const sessionToken = await this.createSessionToken({
+        account,
+        request,
+        tokenVerificationId,
+      });
+      response.sessionToken = sessionToken.data;
+
+      if (query.sendVerifyEmail) {
+        await this.sendVerifyCode({
+          account,
+          request,
+          sessionToken,
+          tokenVerificationId,
+          verificationMethod: 'email-otp',
+        });
+      }
+
+      if (metricsContext?.productId) {
+        const product = (await this.stripeHelper.allProducts()).find(
+          (p) => p.id === metricsContext.productId
+        );
+
+        if (product?.id && product?.name) {
+          await this.subscriptionAccountReminders.create(
+            uid,
+            metricsContext.flowId,
+            metricsContext.flowBeginTime,
+            metricsContext.deviceId,
+            product.id,
+            product.name
```

Comment:
These are required for this `create` method, but `productId` isn't a required property on `metricsContext` (a previously defined schema taken from `/account/create`).

The consequence here is that if we're not provided the optional `productId` on the `metricsContext`, we will quietly not send reminder emails (but the route returns a 200 otherwise).

1. Does this seem like the behavior we want?
2. Is there a way to make `productId` required as part of validation for this route only? (or some better alternative)